beacon-chain/rpc/eth/beacon/handlers.go

```diff
@@ -38,6 +39,934 @@ const (
 	broadcastValidationConsensusAndEquivocation = "consensus_and_equivocation"
 )
 
+type handled bool
+
+// GetBlock retrieves block details for given block ID.
+//
+// DEPRECATED: please use GetBlockV2 instead
+func (s *Server) GetBlock(w http.ResponseWriter, r *http.Request) {
+	ctx, span := trace.StartSpan(r.Context(), "beacon.GetBlock")
+	defer span.End()
+
+	if http2.SszRequested(r) {
+		s.getBlockSSZ(ctx, w, r)
+	} else {
+		s.getBlock(ctx, w, r)
+	}
+}
+
+// getBlock returns the JSON-serialized version of the beacon block for given block ID.
+func (s *Server) getBlock(ctx context.Context, w http.ResponseWriter, r *http.Request) {
+	blockId := mux.Vars(r)["block_id"]
+	if blockId == "" {
+		http2.HandleError(w, "block_id is required in URL params", http.StatusBadRequest)
+		return
+	}
+	blk, err := s.Blocker.Block(ctx, []byte(blockId))
+	if !shared.WriteBlockFetchError(w, blk, err) {
+		return
+	}
```

Comment:
Could we move this part which is exactly the same for both functions into the main `GetBlock` function before the if else? You would need to add `blk` to the parameters but coult remove `r`.
Same comment goes for the other functions below.

Reply:
this change makes sense